cli/src/cmd/utils.rs

```diff
@@ -269,3 +270,24 @@ where
         config
     }
 }
+
+pub fn read_constructor_args_file(constructor_args_path: PathBuf) -> eyre::Result<Vec<String>> {
+    if !constructor_args_path.exists() {
+        eyre::bail!("Constructor args file \"{}\" not found", constructor_args_path.display());
+    }
+    let args = if constructor_args_path.extension() == Some(std::ffi::OsStr::new("json")) {
+        read_json_file(&constructor_args_path).map_err(|e| {
+            eyre::eyre!(
+                "Constructor args file \"{}\" must encode a json array: \"{}\"",
+                constructor_args_path.display(),
+                e
+            )
+        })?
+    } else {
+        fs::read_to_string(constructor_args_path)?
+            .split_whitespace()
+            .map(str::to_string)
```

Comment:
don't think converting al items to String first is required?

Reply:
i think `Vec<String>` can't be built from `std::iter::Iterator<Item=&str>` that `split_whitespace()` returns

Reply:
ah right, this is Vec<>, nvm then